src/magic_folder/test/eliotutil.py

```diff
@@ -69,22 +70,16 @@ def run_and_republish(self, *a, **kw):
         from eliot._output import _DEFAULT_LOGGER as default_logger
 
         def republish():
```

Comment:
There's a newer version of all of this Eliot/testtools integration logic in magic-folder which is both simpler and fixes the interaction with skips - which I'm guessing might have been your motivation here?

If the Tahoe version works here then perhaps it can replace most of what's going on here.  I haven't tried to understand these changes yet because I remember there are so many interactions this code tries to deal with and I'm sure it would take me a long time just to figure out what it was trying to do before, let alone what the effect of the changes are.  If we can get away with the Tahoe version then that would be nice. :)

Reply:
I did have a look at the newer tahoe code, but one thing I noticed about it is that it only add logging to the test method itself, and not any of the setup/teardown/cleanup code, which is something that I was trying to avoid losing.

Reply:
:cry:

Reply:
That said, we could probably adapt some of the tahoe code (the split logger is probably better than the republishing, for example), but I'd rather getting this working now, and deal with adapting the tahoe changes later.

Reply:
Adapting later seems fine to me.  Sorry about the delayed reply.